bolt-sidecar/src/primitives/jsonrpc.rs

```diff
@@ -0,0 +1,73 @@
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use uuid::Uuid;
+
+/// A JSON-RPC request
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct JsonRpcRequest {
+    /// The JSON-RPC version string. MUST be "2.0".
+    pub jsonrpc: String,
+    /// The method string.
+    pub method: String,
+    /// Optional ID.
+    pub id: Option<Value>,
+    /// The parameters object.
+    pub params: Vec<Value>,
+}
+
+/// A JSON-RPC request with a mandatory UUID
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct JsonRpcRequestUuid {
+    /// The JSON-RPC version string. MUST be "2.0".
+    pub jsonrpc: String,
+    /// The method string.
+    pub method: String,
+    /// Optional ID.
+    pub id: Uuid,
+    /// The parameters object.
+    pub params: Vec<Value>,
+}
```

Comment:
why is this needed? can't we reuse Identified<JsonRpcRequest, Uuid> ?

Reply:
Because `Identified` creates a struct with `id` and `inner` field, which doesn't exactly fit a JSON-RPC object. I could have implemented a generic version of `JsonRpcRequest` over `id` but it seemed a bit overkill, also because I want it required in this case.

This made me realize that a procedural macro that adds an "id" field might be better than the `Identified` type, let's see...

Reply:
but since `inner` has a `#[flatten]` attribute it shouldn't matter for deserialization, right?
no need for proc macros IMO :D

Reply:
Hmm true but `JsonRpcRequest` as already an `id` field, as such if you create a struct `Identified<JsonRpcRequest, Uuid>` when serializing you will have two `id` fields. This is how it would be serialized:
```
"{\"jsonrpc\":\"\",\"method\":\"\",\"id\":null,\"params\":[],\"id\":\"019445a4-988d-7eb1-b6c9-444ef07213e8\"}"
```

Reply:
Ah you're right, I didn't notice